x-pack/legacy/plugins/apm/public/components/shared/MetadataTable/index.tsx

```diff
@@ -22,15 +22,35 @@ import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
 import { APMError } from '../../../../typings/es_schemas/ui/APMError';
 import { Span } from '../../../../typings/es_schemas/ui/Span';
 
+type Item = Transaction | APMError | Span;
+
 interface Props {
-  item: Transaction | APMError | Span;
+  item: Item;
   sections: SectionType[];
 }
 
+const filterSections = (sections: SectionType[], item: Item) =>
+  sections.filter(({ key, required, properties }) => {
+    if (required) {
+      return true;
+    }
+    const hasKey = has(item, key);
+    if (!hasKey) {
+      return false;
+    }
+
+    if (properties) {
+      let sectionData: Record<string, unknown> = get(item, key);
+      sectionData = pick(sectionData, properties);
+      if (isEmpty(sectionData)) {
+        return false;
+      }
+    }
+    return true;
+  });
```

Comment:
I think you can centralize all the section logic like this (untested):
```ts
const filterSections = (sections: SectionType[], item: Item) =>
  sections
    .map(section => {
      const data: Record<string, unknown> = get(item, section.key);
      return {
        ...section,
        data: section.properties ? pick(data, section.properties) : data
      };
    })
    .filter(({ required, data }) => required || !isEmpty(data));
```

Then you can simply iterate the valid sections like:
```tsx
      {filteredSections.map(section => {
        return (
          <div key={section.key}>
            <EuiTitle size="xs">
              <h6>{section.label}</h6>
            </EuiTitle>
            <EuiSpacer size="s" />
            <Section propData={section.data} propKey={section.key} />
            <EuiSpacer size="xl" />
          </div>
        );
      })}
```